app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt

```diff
@@ -132,6 +133,42 @@ class HomeScreenRobot {
     fun verifyPrivacyNoticeButton() = assertPrivacyNoticeButton()
     fun verifyStartBrowsingButton() = assertStartBrowsingButton()
 
+    // Upgrading users onboarding dialog
+    fun verifyUpgradingUserOnboardingFirstScreen(testRule: ComposeTestRule) {
+        testRule.also {
+            it.onNodeWithText("Welcome to a more personal internet")
+                .assertIsDisplayed()
+
+            it.onNodeWithText("More colors. Better privacy. Same commitment to people over profits.")
+                .assertIsDisplayed()
+
+            it.onNodeWithText("More colors. Better privacy. Same commitment to people over profits.")
+                .assertIsDisplayed()
```

Comment:
Can we pull from `string.xml` for the above and below?

https://github.com/mozilla-l10n/android-l10n/blob/79f5adafe9a52dab93371f9f51ebae114dd5ec81/mozilla-mobile/fenix/app/src/main/res/values/strings.xml#L271

Reply:
Sure, but I think in some cases it might make sense to have hardcoded text like this to catch regressions if something changes.

Reply:
Done.